src/initialization.js

```diff
@@ -10,31 +10,30 @@ var initialization = {
 */
     initForwarder: function(forwarderSettings, testMode, userAttributes, userIdentities, processEvent, eventQueue, isInitialized) {
         /* `forwarderSettings` contains your SDK specific settings such as apiKey that your customer needs in order to initialize your SDK properly */
-
+        common.settings = forwarderSettings;
         if (!testMode) {
             /* Load your Web SDK here using a variant of your snippet from your readme that your customers would generally put into their <head> tags
                Generally, our integrations create script tags and append them to the <head>. Please follow the following format as a guide:
             */
-
-            // var clientScript = document.createElement('script');
-            // clientScript.type = 'text/javascript';
-            // clientScript.async = true;
-            // clientScript.src = 'https://www.clientscript.com/static/clientSDK.js';   // <---- Update this to be your script
-            // (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(clientScript);
-            // clientScript.onload = function() {
-            //     if (clientSDKObject && eventQueue.length > 0) {
-            //         // Process any events that may have been queued up while forwarder was being initialized.
-            //         for (var i = 0; i < eventQueue.length; i++) {
-            //             processEvent(eventQueue[i]);
-            //         }
-            //          // now that each queued event is processed, we empty the eventQueue
-            //         eventQueue = [];
-            //     }
-            //    clientSDKObject.initialize(forwarderSettings.apiKey);
-            // };
+            var envId = forwarderSettings.environmentId;
+            window.UserLeap = function(){U._queue.push(arguments)}
+            var U = window.UserLeap;
+            U._queue = [];
+            U.envId = envId;
+            var script =document.createElement('script');
+            script.async=1;
+            script.src='https://cdn.userleap.com/shim.js?id='+envId;
+            (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(script);
+            script.onload = function() {
+                isInitialized = true;
+            }
```

Comment:
we queue events in `eventQueue` if UserLeap is not yet available. in order to capture these events, you should include lines 24-33 in this script.onload function.

as I'm looking at the example code on the left, I don't see any UserLeap.initalize method here.  is that not necessary because you are embedding the envId in your script src on line 25?

Reply:
I'm sorry, I don't fully understand how the `eventQueue` related to 24-33. 24-29 is related to setting the <script> properties that loads our js file.

And for initialization, that is handled by our IIFE in our js file. We set the needed properties (envId in 22) in window.UserLeap and everything references that

Reply:
It's very possible that UserLeap has its own concept of an event queue.  We implement one in case our partners don't have one.  Basically, our kit `eventQueue` is abstracted away, but let's say mParticle loads and the user starts firing off events before UserLeap loads.  Unless UserLeap has its own event queue, or you iterate over our provided eventQueue in the kit, these events will get lost.

If UserLeap has its own internal way of handling the event queue then the above comment is moot.

Reply:
Lines 25-31 in the previous commented code show you how to implement our kit `eventQueue`

Reply:
ah got it, yes we have our own event queue that records calls to our SDK before the JS file has been downloaded